elements/lisk-p2p/src/peer.ts

```diff
@@ -628,6 +656,29 @@ export class Peer extends EventEmitter {
 			: {};
 		request.end(legacyNodeInfo);
 	}
+
+	private _banPeer(): void {
+		const banTime =
+			this._peerConfig && this._peerConfig.banTime
+				? this._peerConfig.banTime
+				: DEFAULT_BAN_TIME;
+		setTimeout(this._unbanPeer, banTime);
+		this.disconnect(FORBIDDEN_CONNECTION, FORBIDDEN_CONNECTION_REASON);
+		this.emit(EVENT_BAN_PEER, this._id);
+	}
+
+	private _unbanPeer(): void {
+		this.emit(EVENT_UNBAN_PEER, this._id);
+	}
+
+	private _attachRateInfo = (packet: any) => {
+		packet.rateInfo = {
```

Comment:
prefer not to mutate.
```
return {
...packet,
  rateInfo: {
  }
}
```

Reply:
Problem is what if `packet` has deeply nested fields. This will only do a shallow copy.

Reply:
it's a shallow copy but originally it was mutating directly.
also, why `packet` is any?

Reply:
In this case I used `any` rather than `unknown` to allow for assigning property.  Other option is I can use lodash `merge`